plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AuthenticationError,
+  ConflictError,
+  ForwardedError,
+  InputError,
+  NotFoundError,
+  stringifyError,
+} from '@backstage/errors';
+import { bufferFromFileOrString, KubeConfig } from '@kubernetes/client-node';
+import * as https from 'https';
+import fetch, { RequestInit } from 'node-fetch';
+import { Logger } from 'winston';
+
+import { ClusterDetails, KubernetesClustersSupplier } from '../types/types';
+
+import type { Request } from 'express';
+
+const HEADER_CONTENT_TYPE: string = 'Content-Type';
+const APPLICATION_JSON: string = 'application/json';
+
+const HEADER_KUBERNETES_CLUSTERS: string = 'X-Kubernetes-Clusters';
+
+/**
+ *
+ * @alpha
+ */
+const ERROR_INTERNAL_SERVER: number = 500;
+
+const CLUSTER_USER_NAME: string = 'backstage';
+
+/**
+ *
+ * @alpha
+ */
+export interface KubernetesProxyResponse {
+  code: number;
+  data: any;
```

Comment:
Wondering if this should be `JsonValue | undefined` or something like that (from `@backstage/types`)?

Reply:
I'll check if data is guaranteed to be JSON in this iteration

Reply:
We've scoped the response to JSON for this iteration, however it seems we have a defined known structure in [RFC comment](https://github.com/backstage/backstage/issues/12231#issuecomment-1192875578) where the contents of the `items` correspond to standard kubernetes API responses, I think we can define this response interface with more precision